sample/username_password_sample_with_classification.py

```diff
@@ -0,0 +1,96 @@
+    if cache.has_state_changed else None
+    )
+
+
+# Optional logging
+# logging.basicConfig(level=logging.DEBUG)
+
+config = json.load(open(sys.argv[1]))
+
+# Create a preferably long-lived app instance which maintains a token cache.
+app = msal.PublicClientApplication(
+    config["client_id"], authority=config["authority"], validate_authority=False, token_cache=cache
+    # token_cache=...  # Default cache is in memory only.
+                       # You can learn how to use SerializableTokenCache from
+                       # https://msal-python.rtfd.io/en/latest/#msal.SerializableTokenCache
+    )
+
+# The pattern to acquire a token looks like this.
+result = None
+
+# Firstly, check the cache to see if this end user has signed in before
+accounts = app.get_accounts(username=config["username"])
+print("Accounts"+ str(accounts))
+if accounts:
+    logging.info("Account(s) exists in cache, probably with token too. Let's try.")
+    result = app.acquire_token_silent(config["scope"], account=accounts[0], error_response=True)
+
+if not result:
+    logging.info("No suitable token exists in cache. Let's get a new one from AAD.")
+    # See this page for constraints of Username Password Flow.
+    # https://github.com/AzureAD/microsoft-authentication-library-for-python/wiki/Username-Password-Authentication
+    result = app.acquire_token_by_username_password(
+        config["username"], config["password"], scopes=config["scope"])
+
+if "access_token" in result:
+    # Calling graph using the access token
+    graph_data = requests.get(  # Use token to call downstream service
+        config["endpoint"],
+        headers={'Authorization': 'Bearer ' + result['access_token']}, ).json()
+    print("Graph API call result: %s" % json.dumps(graph_data, indent=2))
+else:
+    option = result.get("classification")
+    if option == "basic_action":
+        print("""Condition can be resolved by user interaction 
+        during the interactive authentication flow.""")
+    elif option == "additional_action":
+        print("""Condition can be resolved by additional remedial interaction
+        with the system, outside of the interactive authentication flow.""")
+    elif option == "message_only":
+        print("""Condition cannot be resolved at this time.
+        Launching interactive authentication flow will show a message 
+        explaining the condition.""")
+        sys.exit("Abort without bothering to try acquire_token_by_xyz()")
+    elif option == "consent_required":
+        print("""Condition can be resolved by giving consent 
+        during the interactive authentication flow.""")
+    elif option == "user_password_expired":
+        print("User password expired")
+    else:
+        print("Default error handling behaviour")
+        result = app.acquire_token_by_username_password(
+            config["username"], config["password"], scopes=config["scope"])
+        print(json.dumps(result, indent=4))
```

Comment:
Shall our sample also demonstrate how to use the returned token to make a graph api call?

Reply:
I can certainly do that. It will however change the structure a little which is fine according to me but if we dont want too many deletions and new additions in the sample, may be we stick to this as calling a graph api is already explained above in the sample. Feel free to suggest another sample structure if you don't want too many additions and deletions.